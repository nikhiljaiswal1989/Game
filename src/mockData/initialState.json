{
  "cardObjectArray": [
    { "isMatchFound": false, "isOpen": false, "text": 60 },
    { "isMatchFound": false, "isOpen": false, "text": 60 },
    { "isMatchFound": false, "isOpen": false, "text": 74 },
    { "isMatchFound": true, "isOpen": false, "text": 56 },
    { "isMatchFound": false, "isOpen": false, "text": 56 },
    { "isMatchFound": false, "isOpen": false, "text": 74 }
  ],
  "clickedIndexArray": [],
  "counter": 0
}
