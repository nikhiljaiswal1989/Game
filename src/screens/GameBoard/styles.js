import { StyleSheet } from 'react-native';
import { color } from '../../constants';
export const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    backgroundColor: color.screenBackground,
    paddingTop: 30,
  },
  board: {
    flexWrap: 'wrap',
    flexDirection: 'row',
  },
  restart: {
    color: color.primaryColor,
    fontSize: 18,
  },
  steps: {
    color: '#fff',
    fontSize: 24,
  },
  counterNumber: {
    color: color.primaryColor,
    fontSize: 32,
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    padding: 24,
  },
});
