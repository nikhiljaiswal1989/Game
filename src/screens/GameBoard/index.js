import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { onClickCard, restartGame } from '../../actions/gameActions';
import { View, Text } from 'react-native';
import { styles } from './styles';
import { Card } from '../../component';
import { CongratulationsAlert } from '../../component';

const GameBoard = () => {
  const { cardObjectArray, counter, isFinish, clickedIndexArray } = useSelector(
    (s) => s.game,
  );
  const dispatch = useDispatch();
  const dispatchOnClickCard = React.useCallback(
    (index) => dispatch(onClickCard(index)),
    [dispatch],
  );
  const dispatchRestart = React.useCallback(() => dispatch(restartGame()), [
    dispatch,
  ]);
  if (isFinish) {
    CongratulationsAlert(counter, dispatchRestart);
  }
  return (
    <View style={styles.mainContainer}>
      <View style={styles.header}>
        <Text
          testID="restart"
          style={styles.restart}
          onPress={() => dispatchRestart()}>
          Restart
        </Text>
        <Text style={styles.steps}>
          STEPS:<Text style={styles.counterNumber}>{counter}</Text>
        </Text>
      </View>
      <View style={styles.board}>
        {cardObjectArray.map((item, index) => {
          return (
            <Card
              testID={'card' + index}
              item={item}
              key={index}
              onClick={() =>
                clickedIndexArray.length < 2 && dispatchOnClickCard(index)
              }
            />
          );
        })}
      </View>
    </View>
  );
};
export default React.memo(GameBoard);
