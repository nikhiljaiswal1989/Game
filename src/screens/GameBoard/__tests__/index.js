import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import GameBoard from '../index';
import { Alert } from 'react-native';
import { render, fireEvent } from '@testing-library/react-native';
import mockData from '../../../mockData/initialState.json';
const mockStore = configureStore();
const store = mockStore({
  game: { ...mockData, isFinish: true },
});

describe('Empty CreateTrip Container ', () => {
  const tree = render(
    <Provider store={store}>
      <GameBoard />
    </Provider>,
  );
  const spy = jest.spyOn(Alert, 'alert');
  it('renders', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('show alert dialog', () => {
    render(
      <Provider store={store}>
        <GameBoard />
      </Provider>,
    );
    Alert.alert.mock.calls[0][2][0].onPress();
    expect(store.getState().game.counter).toEqual(0);
    expect(spy).toHaveBeenCalled();
  });

  it('on Restart press', () => {
    const tree = render(
      <Provider store={store}>
        <GameBoard />
      </Provider>,
    );
    fireEvent.press(tree.getByTestId('restart'));
    expect(store.getState().game.counter).toEqual(0);
  });

  it('on Card press', () => {
    const tree = render(
      <Provider store={store}>
        <GameBoard />
      </Provider>,
    );
    fireEvent.press(tree.getByTestId('card0'));
    expect(store.getState().game.counter).toEqual(0);
  });
});
