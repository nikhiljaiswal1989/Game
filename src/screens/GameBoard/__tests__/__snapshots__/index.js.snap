// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty CreateTrip Container  renders 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(0, 0, 0, 0.8)",
      "flex": 1,
      "paddingTop": 30,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 24,
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "#4AA2EE",
          "fontSize": 18,
        }
      }
      testID="restart"
    >
      Restart
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 24,
        }
      }
    >
      STEPS:
      <Text
        style={
          Object {
            "color": "#4AA2EE",
            "fontSize": 32,
          }
        }
      >
        0
      </Text>
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card0"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 89.98913612301425,
            "transform": Array [
              Object {
                "rotateY": "0.010863876985752086deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              60
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -88.98913612301425,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180.01086387698575deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card1"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 90,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              60
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -89,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card2"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 90,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              74
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -89,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card3"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 90,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              56
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -89,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card4"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 90,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              56
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -89,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="card5"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": 90,
            "transform": Array [
              Object {
                "rotateY": "0deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontSize": 24,
                }
              }
            >
              74
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backfaceVisibility": "hidden",
            "justifyContent": "center",
            "opacity": -89,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "rotateY": "180deg",
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "transparent",
              "borderWidth": 4,
              "height": 296,
              "padding": 4,
              "width": 250,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4AA2EE",
                "borderColor": "#fff",
                "borderRadius": 8,
                "borderWidth": 4,
                "flex": 1,
                "justifyContent": "center",
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 32,
                  "fontWeight": "bold",
                }
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
